traffic_ops/testing/api/v3/servers_test.go

```diff
@@ -26,1017 +26,562 @@ import (
-		}
+func validateServerFieldsForUpdate(hostName string, expectedResp map[string]interface{}) utils.CkReqFunc {
+	return func(t *testing.T, _ toclientlib.ReqInf, _ interface{}, _ tc.Alerts, _ error) {
+		params := url.Values{}
+		params.Set("hostName", hostName)
+		servers, _, err := TOSession.GetServersWithHdr(&params, nil)
+		assert.NoError(t, err, "Error getting Server: %v - alerts: %+v", err, servers.Alerts)
+		assert.Equal(t, 1, len(servers.Response), "Expected Server one server returned Got: %d", len(servers.Response))
+		validateServerFields(expectedResp)(t, toclientlib.ReqInf{}, servers.Response, tc.Alerts{}, nil)
 	}
 }
 
-func GetTestServersIMS(t *testing.T) {
-	var header http.Header
-	header = make(map[string][]string)
-	futureTime := time.Now().AddDate(0, 0, 1)
-	timestamp := futureTime.Format(time.RFC1123)
-	header.Set(rfc.IfModifiedSince, timestamp)
-	params := url.Values{}
-	for _, server := range testData.Servers {
-		if server.HostName == nil {
-			t.Errorf("found server with nil hostname: %+v", server)
-			continue
-		}
-		params.Set("hostName", *server.HostName)
-		_, reqInf, err := TOSession.GetServersWithHdr(&params, header)
-		if err != nil {
-			t.Fatalf("Expected no error, but got %v", err.Error())
-		}
-		if reqInf.StatusCode != http.StatusNotModified {
-			t.Fatalf("Expected 304 status code, got %v", reqInf.StatusCode)
+func validateExpectedServers(expectedHostNames []string) utils.CkReqFunc {
+	return func(t *testing.T, _ toclientlib.ReqInf, resp interface{}, _ tc.Alerts, _ error) {
+		assert.RequireNotNil(t, resp, "Expected response to not be nil.")
+		serverResp := resp.([]tc.ServerV30)
+		var notInResponse []string
+		serverMap := make(map[string]struct{})
+		for _, server := range serverResp {
+			serverMap[*server.HostName] = struct{}{}
+		}
+		for _, expected := range expectedHostNames {
+			if _, exists := serverMap[expected]; !exists {
+				notInResponse = append(notInResponse, expected)
+			}
 		}
+		assert.Equal(t, len(notInResponse), 0, "%d servers missing from the response: %s", len(notInResponse), strings.Join(notInResponse, ", "))
 	}
 }
 
-func CreateTestServers(t *testing.T) {
-	// loop through servers, assign FKs and create
-	for _, server := range testData.Servers {
-		if server.HostName == nil {
-			t.Errorf("found server with nil hostname: %+v", server)
-			continue
-		}
-		resp, _, err := TOSession.CreateServerWithHdr(server, nil)
-		t.Log("Response: ", *server.HostName, " ", resp)
-		if err != nil {
-			t.Errorf("could not CREATE servers: %v", err)
+func validateServerTypeIsNotMid() utils.CkReqFunc {
+	return func(t *testing.T, _ toclientlib.ReqInf, resp interface{}, _ tc.Alerts, _ error) {
+		assert.RequireNotNil(t, resp, "Expected response to not be nil.")
+		serverResp := resp.([]tc.ServerV30)
+		for _, server := range serverResp {
+			if server.Type == tc.CacheTypeMid.String() {
+				t.Errorf("Expected to find no %s-typed servers but found server %s", tc.CacheTypeMid, *server.HostName)
```

Comment:
this will segfault if Traffic Ops responds with a `null` or undefined `hostName`